magenta-lib/src/main/scala/magenta/input/resolver/DeploymentGraphBuilder.scala

```diff
@@ -0,0 +1,46 @@
+package magenta.input.resolver
+
+import magenta.graph.{Edge, EndNode, Graph, MidNode, Node, StartNode}
+import magenta.input.Deployment
+
+object DeploymentGraphBuilder {
```

Comment:
Rather than choosing "three levels" arbitrarily, wouldn't it be better to follow the dependencies, recursively building a graph as you go.